pysaliency/http_models.py

```diff
@@ -20,30 +32,38 @@ def type_url(self):
     
     def conditional_log_density(self, stimulus, x_hist, y_hist, t_hist, attributes=None, out=None):
         # build request
-        pil_image = Image.fromarray(stimulus)
+        stimulus_object = as_stimulus(stimulus)
+
+        # TODO: check for file stimuli, in this case use original file to save encoding time
+        pil_image = Image.fromarray(stimulus_object.stimulus_data)
         image_bytes = BytesIO()
         pil_image.save(image_bytes, format='png')
 
         def _convert_attribute(attribute):
             if isinstance(attribute, np.ndarray):
                 return attribute.tolist()
+            if isinstance(attribute, (np.int64, np.int32)):
+                return int(attribute)
+            if isinstance(attribute, (np.float64, np.float32)):
+                return float(attribute)
             return attribute
 
         json_data = {
-            "x_hist": list(x_hist),
-            "y_hist": list(y_hist),
-            "t_hist": list(t_hist),
+            "x_hist": list(x_hist.tolist()),
```

Comment:
I think if you use `.tolist()`, the `list(...)` is redundant

Reply:
done, just forgot it in the commit